crates/ide_assists/src/handlers/number_representation.rs

```diff
@@ -0,0 +1,183 @@
+use syntax::{ast, ast::Radix, AstToken};
+
+use crate::{AssistContext, AssistId, AssistKind, Assists, GroupLabel};
+
+const MIN_NUMBER_OF_DIGITS_TO_FORMAT: usize = 5;
+
+// Assist: reformat_number_literal
+//
+// Adds or removes seprators from integer literal.
+//
+// ```
+// const _: i32 = 1012345$0;
+// ```
+// ->
+// ```
+// const _: i32 = 1_012_345;
+// ```
+pub(crate) fn reformat_number_literal(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let literal = ctx.find_node_at_offset::<ast::Literal>()?;
+    let literal = match literal.kind() {
+        ast::LiteralKind::IntNumber(it) => it,
+        _ => return None,
+    };
+
+    let text = literal.text();
+    if text.contains('_') {
+        return remove_separators(acc, literal);
+    }
+
+    let (prefix, value, suffix) = literal.split_into_parts();
+    if value.len() < MIN_NUMBER_OF_DIGITS_TO_FORMAT {
+        return None;
+    }
+
+    let radix = literal.radix();
+    let mut converted = prefix.to_string();
+    converted.push_str(&add_group_separators(value, group_size(radix)));
+    converted.push_str(suffix);
+
+    let group_id = GroupLabel("Reformat number literal".into());
+    let label = format!("Convert {} to {}", literal, converted);
+    let range = literal.syntax().text_range();
+    acc.add_group(
+        &group_id,
+        AssistId("reformat_number_literal", AssistKind::RefactorInline),
+        label,
+        range,
+        |builder| builder.replace(range, converted),
+    )
```

Comment:
May I nitpick little bit? :smile:  Things that related to assist application could better reside in `|builder| {... }` closure. Outside are only steps that determine applicability. I.e. application is computed lazily. Doing so reduces _light bulb_ computation complexity.

Reply:
Oh I completely blanked with that when reviewing yes, those things should go inside the closure.

Reply:
Thanks. I'll try to improve it in another mr.

Reply:
@iDawer but I'am using formatted value in the label. Do you think it's better to keep label static, something like "Add digit separators"?

Reply:
Right you are using the formatted string in there(I think the integer base conversion assist does the same). Given that this only applies on longer integer literals anyways without doing a lot of calculations either this is probably fine as is.